### PR TITLE
Start work on defining names and namespaces

### PR DESCRIPTION
**Pull request by cabo**

### CODE REVIEW

File: sdf.md [hunk 1]
```diff
@@ -197,59 +226,127 @@ For example, an Object definition looks like this:
   }
 }
 ~~~
+{: #exobject title="Example Object definition"}
 
-An Object "foo" is defined in the default namespace, containing a property "foo.bar",  of type boolean.
+This example defines an Object "foo" is defined in the default namespace, containing a property "foo.bar",  of type boolean.
```
Comment (akeranen) [inline sdf.md]: s/is defined/that is defined/ ?

File: sdf.md [hunk 2]
```diff
@@ -197,59 +226,127 @@ For example, an Object definition looks like this:
...
+<!-- XXX We are not using the foo.bar style, no? -->
```
Comment (akeranen) [inline sdf.md]: I guess we currently call that simply "odmProperty bar"

Comment (mjkoster) [reply sdf.md]: This is why dot notation is problematic, but I think we could live with name munging as is done in Python. For formal references we should just stick to JSON pointer - odm:/odmObject/foo/odmProperty/bar

File: sdf.md [hunk 3]
```diff
@@ -515,6 +612,9 @@ Modular composition of definitions enables an existing definition (could be in t
 ### Use of the "odmRef" keyword to re-use a definition
 An existing definition may be used as a template for a new definition, that is, a new definition is created in the target namespace which uses the defined qualities of some existing definition. This pattern will use the keyword "odmRef" as a quality of a new definition with a value consisting of a reference to the existing definition that is to be used as a template. Optionally, new qualities may be added and values of optional qualities and quality values may be defined.
 
+ISSUE: Can qualities from the source definition be overridden?
+The above only says "added".
+
```
Comment (mjkoster) [inline sdf.md:617]: We should discuss this to see if there are issues with redefining the values of qualities when using odmRef to create a definition. Maybe we should consider a keyword like "default" to indicate that a quality may be redefined
```
"odmData": 
  "foo" : {
    "type": "number",
    "const": 5,
    "description": "Footacular variable can never be changed"
  }
  "bar" : {
    "type": "number",
    "default": 5,
    "description": "barmatic value may be set in the definition that uses odmRef and redefines the value of bar"
  }

File: sdf.md [hunk 4]
```diff
@@ -197,59 +226,127 @@ For example, an Object definition looks like this:
...
-}
-~~~
+There is no intention that these URIs can be dereferenced.
+(However, as future versions of SDF might find a use for dereferencing
+global names, the URI should be chosen in such a way that this may
+become possible in the future.)
 
-creates a new definition "temperatureProperty" that contains all of the qualities defined in the definition at /odmData/temperatureData.
+The absolute URI of a global name should be a URI as per Section 3 of
+{{-uri}}, with a scheme of "https" and a hier-part.
```
Comment (akeranen) [inline sdf.md]: good to also spell out "hier-part"

File: sdf.md [hunk 5]
```diff
@@ -197,59 +226,127 @@ For example, an Object definition looks like this:
...
+Note that there is no way to provide a URI scheme name in a curie, so
+all references outside of the document need to go through the
+namespace map.
 
-The target namespace is the namespace into which the defined terms are added. The target namespace is defined by the default namespace, or by an explicit prefix on the identifier using a colon ":".
+Name references occur only in specific elements of the syntax of SDF:
 
-For example if the default namespace in the example above is "foo", then you could use "temperatureData" to refer to the property defined at the URI:
+* copying elements via odmRef values
+* pointing to elements via odmRequired value elements
```
Comment (mjkoster) [inline sdf.md:334]: odmInputData, etc.

**Pull request merged**